Client/YooChooseNotifier.php

```diff
@@ -236,93 +235,119 @@ public function hideLocation($locationId, $isChild = false)
-                'action' => 'update',
-                'uri' => $this->getContentUri($content->id),
-                'contentTypeId' => $this->getContentTypeId($content->id),
-            )));
+            $this->notify($notification);
         } catch (RequestException $e) {
-            if (isset($this->logger)) {
-                $this->logger->error('YooChoose Post notification error: ' . $e->getMessage());
-            }
+            $this->log(sprintf('YooChoose Post notification error for unhideLocation: %s', $e->getMessage()), 'error');
         }
     }
 
     /**
-     * Returns ContentType identifier based on $contentId.
-     *
-     * @param int|mixed $contentId
+     * @param string $action
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
+     * @param string|null $lang
      *
-     * @return string
+     * @return array
      */
-    private function getContentTypeIdentifier($contentId)
+    protected function getNotificationContent($action, Content $content, $lang = null)
     {
-        $contentTypeService = $this->contentTypeService;
-        $contentService = $this->contentService;
+        $return = array(
+            'action' => $action,
+            'format' => 'EZ',
+            'uri' => $this->getContentUri($content, $lang),
+            'itemId' => $content->id,
+            'contentTypeId' => $content->contentInfo->contentTypeId,
+        );
 
-        $contentType = $this->repository->sudo(function () use ($contentId, $contentTypeService, $contentService) {
-            $contentType = $this->contentTypeService->loadContentType(
-                $this->contentService
-                    ->loadContent($contentId)
-                    ->contentInfo
-                    ->contentTypeId
-            );
+        if (null !== $lang) {
+            $return['lang'] = $lang;
+        }
+
+        return $return;
+    }
+
+    /**
+     * Checks if content is excluded from supported content types.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
+     *
+     * @return bool
+     */
+    private function isContentTypeExcluded(Content $content)
+    {
+        $contentType = $this->repository->sudo(function () use ($content) {
+            $contentType = $this->contentTypeService->loadContentType($content->contentInfo->contentTypeId);
 
             return $contentType;
         });
 
-        return $contentType->identifier;
+        return !in_array($contentType->identifier, $this->options['included-content-types']);
     }
 
     /**
-     * Gets ContentType ID based on $contentId.
+     * Gets languageCodes based on $content.
      *
-     * @param mixed $contentId
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
+     * @param int|null $versionNo
      *
-     * @return int|null
+     * @return array
      */
-    protected function getContentTypeId($contentId)
+    protected function getLangs($content, $versionNo = null)
```

Comment:
Naming is hard... If you get language codes, I would recommend to have `getLanguageCodes`